src/main/java/fi/aalto/cs/apluscourses/model/SubmissionStatusUpdater.java

```diff
@@ -1,13 +1,17 @@
 package fi.aalto.cs.apluscourses.model;
 
 import com.intellij.notification.Notifications;
+import com.intellij.openapi.project.Project;
 import fi.aalto.cs.apluscourses.intellij.notifications.FeedbackAvailableNotification;
 import fi.aalto.cs.apluscourses.intellij.notifications.Notifier;
 import java.io.IOException;
 import org.jetbrains.annotations.NotNull;
 
 public class SubmissionStatusUpdater {
 
+  @NotNull
+  private final Project project;
```

Comment:
Remark: we've faced some issues with long-living objects with references to IntelliJ's `Project` objects but the lifetime of this object is so short, it shouldn't be an issue here.

Reply:
In this case the reference is luckily gone after at most 3 minutes. Which reminds me that we should add (at some point) some sort of "Failed to get feedback from A+" notification instead of just silently stopping... :thinking: